Reduce database hits

serenasyw opened the pull request.

serenasyw committed:
update models and views with exists()

serenasyw committed:
updated some queries

serenasyw committed:
select_related(), first(), cached_property...

serenasyw committed:
merge reduceSQLHits and reduce number of queries on TeamDetailView

serenasyw committed:
fixed flake8 error

serenasyw committed:
fix the tests

serenasyw committed:
reduce the count(*) queries on admin pages

serenasyw committed:
reduce quries on admin pages

serenasyw committed:
update admin.py and fix flake8 errors

SillyInventor commented on django_teams/models.py:
```diff
@@ -28,67 +29,15 @@ def __unicode__(self):
     def add_user(self, user, team_role=1):
         TeamStatus(user=user, team=self, role=team_role).save()
 
-    def owners(self):
```
If you remove these, you will also need to make a PR to RPI_CSDT_Community to fix the templates when you do (at some point we need to add version control).

SillyInventor commented on django_teams/templates/django_teams/team_list.html:
```diff
@@ -2,10 +2,20 @@
 {% load teams %}
```
This creates duplicate entries for team owners. Which you don't want.

SillyInventor commented on django_teams/templates/django_teams/team_detail.html:
```diff
@@ -12,7 +12,7 @@ <h1>{{ object }} <small>detailed information</small></h1>
             <h3 class="panel-title">Teachers</h3>
```
NotImplementedError at /teams/4/
DISTINCT ON fields is not supported by this database backend

SillyInventor commented on django_teams/templates/django_teams/_table.html:
```diff
@@ -0,0 +1,31 @@
+    <td>
+        {{ object.owner }}
+    </td>
+    <td>{{ object.member_count }}</td>
```
In correctly counts members for classes of which you are the owner (maybe member too?).

SillyInventor commented on django_teams/templates/django_teams/_table.html:
```diff
@@ -0,0 +1,31 @@
+    </td>
+    <td>{% if object.description %}{{ object.description|truncatewords:15 }}{% endif %}</td>
+    <td>
+        {{ object.owner }}
```
In correctly returns owner for the bottom half of the class list.

SillyInventor reviewed (dismissed):
:heart: it. However, there are some problems.

serenasyw committed:
fix incorrect # of member and owner but duplicates exist

serenasyw committed:
remove duplicates in team list

serenasyw committed:
update team list view

serenasyw committed:
update tests

serenasyw committed:
fix travis errors

serenasyw committed:
fix travis errors2

serenasyw committed:
fix travis errors3

serenasyw committed:
add test data

serenasyw committed:
update tests

serenasyw committed:
fix the order

serenasyw committed:
errors solved, and ready to merge

SillyInventor committed:
Merge branch 'master' into reduceDBHits

SillyInventor committed:
Fixing Ownerships

I stopped half way through writing... so it's not really fair for me to
ask you to fix this particular thing.

SillyInventor committed:
Fixing the Details Template

I could have asked you to do this one... but I am anxious for things to
get done

SillyInventor committed:
Fixing Submissions

I needed to go through and fix submitting changes

SillyInventor committed:
Making some general repo changes

SillyInventor committed:
Linting
